runtime/compiler/runtime/IProfiler.cpp

```diff
@@ -4290,26 +4290,26 @@ TR_IPHashedCallSite::operator new (size_t size) throw()
 inline
 uintptrj_t CallSiteProfileInfo::getClazz(int index)
    {
-#if defined(OMR_GC_COMPRESSED_POINTERS) //compressed references
-   //support for convert code, when it is implemented, "uncompress"
-   return (uintptrj_t)TR::Compiler->cls.convertClassOffsetToClassPtr((TR_OpaqueClassBlock *)(uintptrj_t)_clazz[index]);
-#else
-   return (uintptrj_t)_clazz[index]; //things are just stored as regular pointers otherwise
-#endif //OMR_GC_COMPRESSED_POINTERS
+   if (TR::Compiler->om.compressObjectReferences())
+      //support for convert code, when it is implemented, "uncompress"
+      return (uintptrj_t)TR::Compiler->cls.convertClassOffsetToClassPtr((TR_OpaqueClassBlock *)(uintptrj_t)_clazz[index]);
+   else
+      return (uintptrj_t)_clazz[index]; //things are just stored as regular pointers otherwise
    }
 
 inline
 void CallSiteProfileInfo::setClazz(int index, uintptrj_t clazzPointer)
    {
-#if defined(OMR_GC_COMPRESSED_POINTERS) //compressed references
-   //support for convert code, when it is implemented, do compression
-   TR_OpaqueClassBlock * compressedOffset = J9JitMemory::convertClassPtrToClassOffset((J9Class *)clazzPointer); //compressed 32bit pointer
-   //if we end up with something in the top 32bits, our compression is no good...
-   TR_ASSERT((!(0xFFFFFFFF00000000 & (uintptrj_t)compressedOffset)), "Class pointer contains bits in the top word. Pointer given: %p Compressed: %p", clazzPointer, compressedOffset);
-   _clazz[index] = (uint32_t)((uintptrj_t)compressedOffset); //ditch the top zeros
-#else
-   _clazz[index] = (uintptrj_t)clazzPointer;
-#endif //OMR_GC_COMPRESSED_POINTERS
+   if (TR::Compiler->om.compressObjectReferences())
+      {
+      //support for convert code, when it is implemented, do compression
+      TR_OpaqueClassBlock * compressedOffset = J9JitMemory::convertClassPtrToClassOffset((J9Class *)clazzPointer); //compressed 32bit pointer
+      //if we end up with something in the top 32bits, our compression is no good...
+      TR_ASSERT((!(0xFFFFFFFF00000000 & (uintptrj_t)compressedOffset)), "Class pointer contains bits in the top word. Pointer given: %p Compressed: %p", clazzPointer, compressedOffset);
+      _clazz[index] = (uint32_t)((uintptrj_t)compressedOffset); //ditch the top zeros
```

Comment:
The type of `_clazz` is now an array of `uintptrj_t` rather than `uint32_t` like it used to be. But it's unsigned so the store here and the load in the getter above should still be valid. Just noting I've verified this.

Reply:
I believe the code is correct with the larger type, and given the cache size here is 3, I don't think this will waste much space in the normal builds (i.e. it wasn't worth trying to keep the type as uint32_t in the compressed-only build).